modules/mobile/spec/request/preferred_name_request_spec.rb

```diff
@@ -7,53 +7,115 @@
 RSpec.describe 'preferred_name', type: :request do
   include SchemaMatchers
 
-  let(:user) { FactoryBot.build(:iam_user) }
+  describe 'logingov user' do
+    let(:login_uri) { 'LGN' }
 
-  before do
-    iam_sign_in(user)
-    allow_any_instance_of(VAProfile::Demographics::Service).to receive(:identifier_present?).and_return(true)
-  end
+    before do
+      iam_sign_in(FactoryBot.build(:iam_user, :logingov))
```

Comment:
In some of the tests below, you also set the idme_id to nil for these logingov users. Is there a reason that we're not doing the stubs on the instance anyhow, like: 
```
let(:user) { FactoryBot.build(:iam_user, :logingov) }
allow(user).to receive(:logingov_uuid).and_return('b2fab2b5-6af0-45e1-a9e2-394347af91ef')
```

Reply:
Ya that's a fair point, I'll swap it to stub the instance

Reply:
Actually I don't think that's possible. The instance created in the controller is different from the instance created in the spec